src/theory/quantifiers/conjecture_generator.cpp

```diff
@@ -1089,16 +1089,17 @@ void ConjectureGenerator::getEnumerateUfTerm( Node n, unsigned num, std::vector<
         vec_sum = 0;
         vec.push_back( size_limit );
       }else{
+        int vecindex = (index < vec.size()) ? vec[index] : 0;
         //see if we can iterate current
         if (vec_sum < size_limit
-            && !te->getEnumerateTerm(types[index], vec[index] + 1).isNull())
+            && !te->getEnumerateTerm(types[index], vecindex + 1).isNull())
         {
-          vec[index]++;
+          vec.assign(index, vecindex + 1);
```

Comment:
My understanding of std::vector assign is that this would update vec so that it contained index elements, all equal to vecindex+1 (see http://www.cplusplus.com/reference/vector/vector/assign/).  Perhaps this should be:
```
if( index<vec.size() ) 
{
  vec[index]++;
}
```